frontend/src/metabase/components/Popover/Popover.jsx

```diff
@@ -240,14 +240,11 @@ export default class Popover extends Component {
       >
         {typeof this.props.children === "function"
           ? this.props.children(childProps)
-          : React.Children.count(this.props.children) === 1 &&
-            // NOTE: workaround for https://github.com/facebook/react/issues/12136
-            !Array.isArray(this.props.children)
-          ? React.cloneElement(
-              React.Children.only(this.props.children),
-              childProps,
-            )
-          : this.props.children}
+          : React.Children.map(this.props.children, child =>
+              React.isValidElement(child)
+                ? React.cloneElement(child, childProps)
```

Comment:
This issue seems to be caused by https://github.com/metabase/metabase/pull/25708. Historically, we didn't think it gonna cause any issues. However, due to the fact that we need a custom footer on the map options
<img src="https://user-images.githubusercontent.com/1328979/193100158-4f2e4e70-36d7-4dcd-beb4-3f85e6f89b37.png" />

currently, we pass along `footer` together with the `Accordion` that's why it doesn't get the `maxHeight` prop.

https://github.com/metabase/metabase/blob/ad8a8982ec3f053dd156b13c16560c539739a33e/frontend/src/metabase/core/components/Select/Select.tsx#L325

However, in that case, we'll also pass along `maxHeight` to the footer which doesn't make sense.
<img width="1099" alt="image" src="https://user-images.githubusercontent.com/1937582/221199603-16aa04f8-3e39-4f42-80ae-5b27f78be91d.png">

I manually check every place used by `Popover` and only 3 have passed more than 1 child.

1. DataSelector: Although, keyboard arrows don't work, <kbd>Tab</kbd> navigation works fine.
    <img width="361" alt="image" src="https://user-images.githubusercontent.com/1937582/221200110-0c25a44e-4ce2-40f1-b3bb-ef52bb8ba566.png">
2. TimeseriesFilterWidget: It's so short it shouldn't be a problem.
<img width="484" alt="Screenshot 2023-02-24 at 1 32 31 PM" src="https://user-images.githubusercontent.com/1937582/221200277-b24b4966-8b43-460f-876f-1d3bbcd1ab2c.png">
3. Select:  which this PR attempts to fix

I proposed you worked with @npfitz to come up with an API that will fix the `Select` component that allows it to pass additional `footer` elements that won't break the accordion `maxHeight` prop. I've try to see if it's possible to use `sections` props in `AccordionList` to receive the `footer`, but seems `sections` expect `items` key which would be selected by users. However, I didn't give it a more thorough investigation.